src/routers/observations.py

```
@@ -0,0 +1,171 @@
+    except ValueError as e:
+        logger.warning(f"Failed to get observations for session {session_id}: {str(e)}")
+        raise ResourceNotFoundException("Session not found") from e
+
+
+@router.post("/query", response_model=list[schemas.Observation])
+async def query_observations(
+    workspace_id: str = Path(..., description="ID of the workspace"),
+    session_id: str = Path(..., description="ID of the session"),
+    body: schemas.ObservationQuery = Body(
+        ..., description="Semantic search parameters for observations"
+    ),
```

Ruff (0.14.2)

Do not perform function call `Body` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)